pyenvisalink/dsc_client.py

```diff
@@ -207,3 +224,29 @@ def handle_keypad_update(self, code, data):
         for part in self._alarmPanel.alarm_state['partition']:
             self._alarmPanel.alarm_state['partition'][part]['status'].update(new_status)
         _LOGGER.debug(str.format("(All partitions) state has updated: {0}", json.dumps(new_status)))
+
+    def handle_zone_bypass_update(self, code, data):
+        """ Handle zone bypass update triggered when *1 is used on the keypad """
+        self._refreshZoneBypassStatus = False
+        if len(data) == 16:
+            for byte in range(8):
+                bypassBitfield = int('0x' + data[byte * 2] + data[(byte * 2) + 1], 0)
+
+                for bit in range(8):
+                    zoneNumber = (byte * 8) + bit + 1
+                    bypassed = (bypassBitfield & (1 << bit) != 0)
+                    self._alarmPanel.alarm_state['zone'][zoneNumber]['bypassed'] = bypassed
+                    _LOGGER.debug(str.format("(zone {0}) bypass state has updated: {1}", zoneNumber, bypassed))
+
+        else:
+            _LOGGER.error(str.format("Invalid data length ({0}) has been received in the bypass update.", len(data)))
+
+    async def dump_zone_bypass_status(self):
```

Comment:
I think this whole thing here can be done in a much more simple way (if I'm understanding this correctly).  It appears that this is setup to poll every 5 seconds, but the only time it will actually do anything - which is to send *1# to the keypad, is at initial login, and also when the partition is disabled.

Can't we just add: ```self.keypresses_to_partition(1, "*1#")``` at the end of ```handle_login_success``` and also within the ```handle_partition_state_change``` event (if you add the 655 processing to it as mentioned above.  Just hoping to make it simpler if it still works!